.htaccess

```diff
@@ -1,240 +1,561 @@
 
-    <Files RELEASE_NOTES.txt>
-        order allow,deny
-        deny from all
-    </Files>
+	#RewriteRule ^(.*)$ https://www.example.com/ [L,R=301]
 
-############################################
-## If running in cluster environment, uncomment this
-## http://developer.yahoo.com/performance/rules.html#etags
+</Files>
+
+<FilesMatch "\.(all-contributorsrc|git.*/?)">
+	
+###########################################
+## Deny access (Default)
+## Comment the lines if you use the alternative method
 
-    #FileETag none
+	Order allow,deny
+	Deny from all
 
 ###########################################
-## Deny access to cron.php
-    <Files cron.php>
+## Redirect to external location (Alternative)
+## Apache Module mod_rewrite must be enabled
+
+	#RewriteRule ^(.*)$ https://www.example.com/ [L,R=301]
+
+</FilesMatch>
+
+<FilesMatch "(.*)\.(json|lock|md|sh|txt|sample)$">
+
+###########################################
+## Deny access (Default)
+## Comment the lines if you use the alternative method
+	
+	Order allow,deny
+	Deny from all
+
+###########################################
+## Redirect to external location (Alternative)
+## Apache Module mod_rewrite must be enabled
+	
+	#RewriteRule ^(.*)$ https://www.example.com/ [L,R=301]
+
+</FilesMatch>
+
+<IfModule mod_setenvif.c>
 
 ############################################
-## uncomment next lines to enable cron access with base HTTP authorization
-## http://httpd.apache.org/docs/2.2/howto/auth.html
+## Block abusive bot traffic to reduce the performance impact
 ##
-## Warning: .htpasswd file should be placed somewhere not accessible from the web.
-## This is so that folks cannot download the password file.
-## For example, if your documents are served out of /usr/local/apache/htdocs
-## you might want to put the password file(s) in /usr/local/apache/.
+## Create an alias for this command and check periodically the Apache access log file for bots
+##    awk -F\" '/GET/ {print $6}' /var/log/apache2/access.log | sort | uniq -c | sort -fn | grep -wviE '(Android|iPad|iPhone|Macintosh|Windows NT|X11)'
+## In AWStats check Robots/Spiders visitor page
+## https://www.the-art-of-web.com/system/logs/
+##
+## Limit the REGEX expression to a few words and do not mention the bot version
+## For information about known bots visit https://user-agents.net/bots
+
+	BrowserMatchNoCase "^19(.*)" BAD_BOT
+	BrowserMatchNoCase "adidxbot(.*)" BAD_BOT
+	BrowserMatchNoCase "AhrefsBot(.*)" BAD_BOT
+	BrowserMatchNoCase "ApacheBench(.*)" BAD_BOT
+	BrowserMatchNoCase "BlackWidow(.*)" BAD_BOT
+	BrowserMatchNoCase "BLEXBot(.*)" BAD_BOT
+	BrowserMatchNoCase "^Bot(.*)" BAD_BOT
+	BrowserMatchNoCase "CensysInspect(.*)" BAD_BOT
+	BrowserMatchNoCase "CFNetwork(.*)" BAD_BOT
+	BrowserMatchNoCase "ChinaClaw(.*)" BAD_BOT
+	BrowserMatchNoCase "Crawlson(.*)" BAD_BOT
+	BrowserMatchNoCase "coccocbot(.*)" BAD_BOT
+	BrowserMatchNoCase "cortex(.*)" BAD_BOT
+	BrowserMatchNoCase "^Custo(.*)" BAD_BOT
+	BrowserMatchNoCase "^curl(.*)" BAD_BOT
```

Comment:
I think this whole block is too opinionated... What if someone has some process they're using with curl that would now be blocked?

Reply:
As far as curl is debatable, that piece is taken from an online bad bots generator and that has curl included. If the directive is not commented the curl access from the terminal will be blocked if you do not establish a User-Agent. Daily I am looking at the bots list and I found that around 700 hits are through the curl. As it is not made by the team I work in we block bots that have User-Agent that starts with curl.

Reply:
It being debatable is more or less what I mean by opinionated.

Curl may not be a good bot for you, but some poor soul probably has some script running in the office closet making curl requests to their magento site to run some process. If it suddenly stops working they may not know why, especially if it was created by some developer long gone.

Also, ahrefs bot. Some people probably use ahref's marketing tools that would now break. Ahrefs claims they're a good bot: https://ahrefs.com/robot since they obey robots.txt etc. But just because you do not use it, we should not block it from everyone's openmage site.

Maybe it's better to remove all the user agents and just leave the awk command that users can generate for themselves.

Reply:
The initial list was created by an online generator. I checked all of them with the help of the website mentioned on a line in the proposed htaccess. Then for two 2 months I added all the bots that generated high daily traffic, here I do not mean 10 hits but over 100. I analyze the objection related to curl, I will comment on that line. I will add Ahrefs in robots.txt, if it continues to crawl my website I will keep that line.

Reply:
Then maybe it's better to just link the online generator so people can decide themselves. If other reviewers are okay with a set of defaults then okay, I just think it's too opinionated.

Reply:
This list has too many false positives, ahrefs is an extremely famous seo tool, ApacheBench is a benchmarking tool, Dalvik is the android vmachine name, wget and curl are highly debatable.